Add two more specific error codes for path abandon frame (#466)
quicwg/multipath: draft-ietf-quic-multipath.md

limitations in the transport layer's capacity. This error indicates that
resource constraints prevent the continuation of the path.

UNSTABLE_INTERFACE (TBD-11): The endpoint is abandoning the path because

> of unstable interfaces. This error is used when endpoints find that

Reviewer: Should this be interfaces in singular?

the network interfaces are unstable due to weak signal or other possible situations.

> Endpoints could also choose to use this error when detecting a black hole on the specific path.

Reviewer: I think black hole can be replace by very high degrees of persistent packet loss, rather than using jargon.

limitations in the transport layer's capacity. This error indicates that
resource constraints prevent the continuation of the path.

UNSTABLE_INTERFACE (TBD-11): The endpoint is abandoning the path because

> the used interface is considered to be unstable. This conition can occur, e.g.,

Reviewer 2: ```suggestion
the used interface is considered to be unstable. This condition can occur, e.g.,
```

> the used interface is considered to be unstable. This conition can occur, e.g., 

Reviewer 3: ```suggestion
the used interface is considered to be unstable. This conition can occur, e.g.,
```

> due to a weak wireless signal or frequent mobility events.

Reviewer 4: What does "frequent mobility events" mean here? Maybe consider making this more clear.

Reviewer 3: This is e.g. when you walk constantly in and out of coverage. However, these are really just supposed to be examples, so I don't think we need to go too much into detail. Which part is not clear? Do you have an alternative wording that you think would be more clear?

Author: ```suggestion
due to a weak wireless signal or frequent handover events during high-speed mobility.
```

NO_CID_AVAILABLE (TBD-12): The endpoint is abandoning the path due to
the lack of a connection ID for this path.
This may occur when the peer initiates a new path
but has not provided a corresponding connection ID for the path ID

> (or the packet containing the connection IDs has not arrived yet).

Reviewer: Should this error position itself in relation to the PATH_CID_BLOCKED in #468. Commonly a endpoint would first send PATH_CID_BLOCKED to indicate the lack of CID, and eventually it might decide to close the path using PATH_ABANDON with the NO_CID_AVAILABLE error code.

Author: Yes, that's true. But we need to merge #468 first or we'll have an incorrect reference, and we also need to change the error code TBD-number at that point. I'd like to change these two parts after #468 is merged.

Reviewer 3: Not sure if an endpoint would typically do that. It's is equally valid to just abandon the path right away and not have an half-open path dangling around. I'm not sure we want to give any guidance here at all.

Reviewer: After some additional thought I think this is fine. Implementers will likely figure out what I wrote, or if not even that will be clear for the ones receiving PATH_ABANDON with this error code.

limitations in the transport layer's capacity. This error indicates that
resource constraints prevent the continuation of the path.

> UNSTABLE_INTERFACE (TBD-11): The endpoint is abandoning the path because
> of unstable interfaces. This error is used when endpoints find that
> the network interfaces are unstable due to weak signal or other possible situations.
> Endpoints could also choose to use this error when detecting a black hole on the specific path.

Reviewer 3: ```suggestion
UNSTABLE_INTERFACE (TBD-11): The endpoint is abandoning the path because
the used interface is considered to be unstable. This conition can occur, e.g., 
due to a weak wireless signal or frequent mobility events.
```
